contrib/endpoints/src/api_manager/service_control/aggregated.cc

```diff
@@ -497,56 +537,17 @@ void Aggregated::Call(const RequestType& request, ResponseType* response,
     max_report_size_ = request_body.size();
   }
 
-  auto token = (typeid(RequestType) == typeid(AllocateQuotaRequest))
-                   ? sa_token_quota_
-                   : sa_token_;
-
   http_request->set_url(url)
       .set_method("POST")
-      .set_auth_token(GetAuthToken(token))
+      .set_auth_token(GetAuthToken(request))
       .set_header("Content-Type", application_proto)
       .set_body(request_body);
 
-  // Set timeout on the request if it was so configured.
-  if (typeid(RequestType) == typeid(CheckRequest)) {
-    http_request->set_timeout_ms(kCheckDefaultTimeoutInMs);
-  } else if (typeid(RequestType) == typeid(AllocateQuotaRequest)) {
-    http_request->set_timeout_ms(kAllocateQuotaDefaultTimeoutInMs);
-  } else {
-    http_request->set_timeout_ms(kReportDefaultTimeoutInMs);
-  }
-
-  if (server_config_ != nullptr &&
-      server_config_->has_service_control_config()) {
-    const auto& config = server_config_->service_control_config();
-    if (typeid(RequestType) == typeid(CheckRequest)) {
-      if (config.check_timeout_ms() > 0) {
-        http_request->set_timeout_ms(config.check_timeout_ms());
-      }
-    } else if (typeid(RequestType) == typeid(AllocateQuotaRequest)) {
-      if (config.quota_timeout_ms() > 0) {
-        http_request->set_timeout_ms(config.quota_timeout_ms());
-      }
-    } else {
-      if (config.report_timeout_ms() > 0) {
-        http_request->set_timeout_ms(config.report_timeout_ms());
-      }
-    }
-  }
+  SetHttpRequestTimeout(request, http_request);
```

Comment:
It is better:

auto request_timeout = GetHTTPRequestTimeout<RequestType>();
http_request->set_timeout_ms(request_timeout);